csrc/cpu/comm/ccl.cpp

```diff
@@ -526,7 +534,10 @@ void inference_all_reduce(torch::Tensor& data, py::object op, py::object group,
         workspace[world_rank].state = 0;
     }
     if (world_rank != 0) {
-        wait_buffer_state_until(0, 0);
+        // if rank 0 spin too fast it could be in state 1 of next allreduce
+        // in this case wait_buffer_state_until(0, 0) may cause deadlock
+        // what we are certain is when rank 0 finishes the state won't be 2
+        wait_buffer_state_until_not(0, 2);
```

Comment:
Can we please represent the states with symbolic constants with descriptive names to enhance readability and maintenance? Thanks!

Reply:
state changed to enum type.  Thanks for the suggestion!